feat: improve wallet connect variables check

RyukTheCoder opened the pull request:
# Summary

Implemented throwing error if WalletConnect is not excluded and WC_PROJECT_ID have not passed to provider-all.

Fixes # 1464


# How did you test this change?

Tested it by excluding and including wallet-connect in widget wallets list and also adding and removing WC_PROJECT_ID.


# Checklist:

- [x] I have performed a self-review of my code

yeager-eren commented on wallets/provider-all/src/helpers.ts:
```diff
@@ -0,0 +1,12 @@
+  selectedProviders?: (WalletType | ProviderInterface)[]
+) =>
+  selectedProviders &&
+  !selectedProviders.find(
```
You are only checking string mode of the union (`selectedProviders`).

I guess it should be something like this:
```ts
  !selectedProviders.find((provider) => {
    if (typeof provider === 'string') {
      return provider === WalletTypes.WALLET_CONNECT_2;
    }
    return provider.getWalletInfo([]).name;
  });
```

RyukTheCoder replied:
Tnx, Fixed

yeager-eren commented on wallets/provider-all/src/index.ts:
```diff
@@ -28,10 +32,26 @@ import * as trustwallet from '@rango-dev/provider-trustwallet';
-  walletconnect2.init(enviroments?.walletconnect2 || {});
+export const allProviders = (
+  environments?: Environments,
+  selectedProviders?: (WalletType | ProviderInterface)[]
```
I think you don't need to make a separate param. you can add it to `environments`. you can rename `environments` to `options`. I'm okay with both approach.

yeager-eren reviewed (changes requested):
Thanks. I've added some comments on your code. Please ask @samobasquiat to check it also.

yeager-eren reviewed (approved):
LGTM. please ask @samobasquiat to do a second review also.

samobasquiat commented:
Great job.
I think we should also consider making the `config` parameter in the `useWalletList` function required.

RyukTheCoder commented:
> Great job. I think we should also consider making the `config` parameter in the `useWalletList` function required. For example, recently, I fixed a bug related to this hook with the changes in https://github.com/rango-exchange/rango-client/pull/683/files which we missed due to the absence of type guarding.

I didn't get the point of `config` being required and getting it from `useAppStore` and passing it to  `useWalletList` everywhere we want to use it, so I moved getting config from `useAppStore` inside the `useWalletList` hook.
